packages/medusa/src/services/csv-parser.ts

```diff
@@ -0,0 +1,95 @@
+import { AwilixContainer } from "awilix"
+import {
+  AbstractParser,
+  LineContext,
+  ParserOptions,
+} from "../interfaces/abstract-parser"
+import { CsvSchema, ICsvParser } from "../interfaces/csv-parser"
+import Papa, { ParseConfig } from "papaparse"
+import fs from "fs"
+
+class CsvParser<
+    TSchema extends CsvSchema = CsvSchema,
+    TParserResult = any,
+    TOutputResult = any
+  >
+  extends AbstractParser<TSchema, TParserResult, TOutputResult>
+  implements ICsvParser<TParserResult>
+{
+  $$delimiter = ";"
+
+  constructor(
+    protected readonly container: AwilixContainer,
+    schema: TSchema,
+    delimiter?: string
+  ) {
+    super(schema)
+    if (delimiter) {
+      this.$$delimiter = delimiter
+    }
+  }
+
+  public async parse(
+    path: string,
```

Comment:
question: should we add a method parseFromStream as well?

Reply:
comment: if we add a method parseFromStream then this method implementation will only create a stream from the file and return await this.parseFromStream(fileStream, options)

Reply:
I agree, I think that to make this as generalized as possible we should have a method that takes in a readable and consumes it as done with the `fileStream` after `fs.createReadStream`. We could provide a path as well as done here but I don't think it's strictly necessary, otherwise the readable can be instantiated outside the csvparser